### ISSUE TITLE
Dock remains visible after closing overview when "Show the Dock on Mouse Hover" is disabled

### ISSUE THREAD

**Issue by jderose9** [passingthru67/workspaces-to-dock#81]
###Linux Distribution version
Fedora 25

###Gnome Shell version
3.22.2

###Extension version or branch
40

###Description of the problem
When "Intelligent Hiding > Dock is Fixed" and "Autohide > Show the Dock on Mouse Hover" are both off (basically I only want to show when in the overview or when the keystroke is pressed), the dock remains visible after closing the overview.

###Steps to reproduce the problem
1. Turn off "Intelligent Hiding > Dock is Fixed" and "Autohide > Show the Dock on Mouse Hover"
2. Open Overview (with super key)
3. Close overview (with super key)
4. Dock is still visible even after the overview closes.

**Comment by passingthru67** [comment]
Have you tried turning off the Intellihide feature? Intellihide will show the dock unless an app window overlaps it.

**Comment by passingthru67** [comment]
Any update on this issue?

**Comment by jderose9** [comment]
The positive/negative cases are a bit confusing to me, but here are my settings:

![screenshot from 2017-06-01 21-22-31](https://cloud.githubusercontent.com/assets/2118702/26707170/cf63a230-4710-11e7-8a37-63130a17375f.png)

With these settings, the dock does not appear when the extension loads until I open the overview, then it gets stuck open.

If I flip the first switch to ON, it will remain visible always. My hope is to set it such that it will open/close when the overview opens/closes, and that I can additionally open/close it via keystroke, but mouseover and window sizes do not trigger it.

**Comment by passingthru67** [comment]
Try this .. export your settings by opening a terminal and typing "dconf dump /org/gnome/shell/extensions/workspaces-to-dock/". Then, copy/paste the results here. I'll try to duplicate your environment.

**Comment by passingthru67** [comment]
Also, do you have any other extensions installed?

**Comment by jderose9** [comment]
Thanks! here you go:
```
[/]
shortcuts-panel-show-running=true
workspace-caption-taskbar-icon-size=24.0
dashtodock-hover=false
shortcuts-panel-orientation='OUTSIDE'
dock-fixed=false
intellihide=false
hide-dash=false
customize-height=true
quick-show-timeout=600.0
overview-action='SHOW_FULL'
animation-time=0.14999999999999999
shortcuts-panel-appsbutton-at-bottom=false
shortcuts-panel-show-window-count-indicators=true
hide-delay=0.0
shortcuts-panel-appsbutton-animation=false
autohide=false
customize-height-option=0
shortcuts-panel-icon-size=48.0
require-pressure-to-show=true
toggle-dock-with-keyboard-shortcut=true
dock-position='RIGHT'
quick-show-on-workspace-change=false
show-delay=0.0
shortcuts-panel-show-places=false
opaque-background=false
show-shortcuts-panel=false
top-margin=0.0
bottom-margin=0.0
toggle-overview=true
```

I wanted to mention that after opening the overview, it gets stuck open even if over top of a focused application window, rather than exhibiting the intellihide behavior.

**Comment by passingthru67** [comment]
This issue has been resolved in the latest gnome-3.22 branch commits. I'm currently working on backporting other fixes for Gnome 3.22 and I plan on uploading a new version to extensions.gnome.org when I'm done.

In the mean time, please download the gnome-3.22 branch zip (https://github.com/passingthru67/workspaces-to-dock/archive/gnome-3.22.zip) and extract it into a temporary folder. Inside of the extracted folder you will find the workspaces-to-dock@passingthru67.gmail folder. Simply copy it to your ~/.local/share/gnome-shell/extensions folder. Next, restart Gnome Shell (alt+f2+r).

Let me know how it turns out.
